validator/db/kv/attester_protection.go

```diff
@@ -441,25 +441,41 @@ func (s *Store) saveAttestationRecords(ctx context.Context, atts []*AttestationR
 			if err != nil {
 				return errors.Wrap(err, "could not create public key bucket")
 			}
+
 			sourceEpochBytes := bytesutil.EpochToBytesBigEndian(att.Source)
 			targetEpochBytes := bytesutil.EpochToBytesBigEndian(att.Target)
 
 			signingRootsBucket, err := pkBucket.CreateBucketIfNotExists(attestationSigningRootsBucket)
 			if err != nil {
 				return errors.Wrap(err, "could not create signing roots bucket")
 			}
+
+			if s.slashingProtectionType == minimal {
+				if err := emptyBucket(signingRootsBucket); err != nil {
+					return errors.Wrap(err, "could not empty signing roots bucket")
+				}
+			}
+
 			if err := signingRootsBucket.Put(targetEpochBytes, att.SigningRoot); err != nil {
 				return errors.Wrapf(err, "could not save signing root for epoch %d", att.Target)
 			}
+
 			sourceEpochsBucket, err := pkBucket.CreateBucketIfNotExists(attestationSourceEpochsBucket)
 			if err != nil {
 				return errors.Wrap(err, "could not create source epochs bucket")
 			}
 
+			if s.slashingProtectionType == minimal {
```

Comment:
There is a theme here that is confusing. The minimal db would keep a `sourceEpochsBucket` and a `targetEpochBucket` that is deleted on each attestation and the last attestation is recorded. This in itself keeps repeated information as both buckets point to the exact same attestation data. 

Later below the buckets for `lowestSignedSourceBucket` and `lowestTargetBucket` is also filled with the same attestation data yet again. 

Why can't we simply then ignore this whole block here for the first two buckets and only restrict ourselves to using the last two buckets in case of minimal db?

Reply:
Well, I tried during a few minutes to find a counter example proving why your solution won't work.
I cannot. So, I guess you'r right.